adapters/adpone/adpone.go

```diff
@@ -0,0 +1,116 @@
+package adpone
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+)
+
+func NewAdponeBidder(endpoint string) *adponeAdapter {
+	return &adponeAdapter{endpoint: endpoint}
+}
+
+type adponeAdapter struct {
+	endpoint string
+}
+
+func (adapter *adponeAdapter) MakeRequests(
+	openRTBRequest *openrtb.BidRequest,
+	reqInfo *adapters.ExtraRequestInfo,
+) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+	var imp = &openRTBRequest.Imp[0]
```

Comment:
When a request with an empty `openRTBRequest.Imp` array comes in, your adapter fails with an "index out of range" error in line 29. Can we check for `nil` before dereferencing the `Imp` array and check its length before accessing any of its elements?